src/stats/mod.rs

```diff
@@ -27,23 +34,117 @@ pub struct GpuStats {
     pub memory_total_kb: u64, // sum of total memory across all GPUs
 }
 
-pub fn get_num_cpus() -> Option<f64> {
-    let source = get_best_system_stats_source_for(ResourceType::NumCpus)?;
+#[derive(Debug, Eq, PartialEq, Clone)]
+pub enum CgroupVersion {
+    V1,
+    V2,
+}
+
+pub fn get_num_cpus(
+    cgroup_version: Option<CgroupVersion>,
+    cgroup_v2_mount_point: Option<PathBuf>,
+    cgroup_v1_mount_points: Option<CgroupV1MountPoints>,
+) -> Option<f64> {
+    if Some(CgroupVersion::V2) == cgroup_version {
+        if let Some(v2_mount_point) = cgroup_v2_mount_point {
```

Comment:
Why would we have `CgroupVersion::V2`, but then no V2 mount point, or vice versa? (Consider this repeated.)

Reply:
shouldn't happen, it should always exist, unless there is a missing logic for resolving it from the `/proc/mounts`

all-in-all I don't like this particularly repeat that I introduced as I moved the path resolution to the outer scope, and was thinking of DRYing this all to `get_resource_stat(enum)` or something 🤔

the only thing that changes is the functions they call on the sources

Reply:
for v1 it's a bit more complicated as some mount points might exist but not the one needed for this particular resource (cpu, cpuacct, memory)

but that is handled deeper in the code for those

Reply:
I, the very least, DRYed the stat reading priority order!